app/models/manageiq/providers/ansible_tower/inventory/parser/automation_manager.rb

```diff
@@ -32,4 +33,20 @@ def configuration_scripts
       o[:inventory_root_group] = target.inventory_groups.lazy_find(i.inventory_id.to_s)
     end
   end
+
+  def configuration_script_sources
+    collector.projects.each do |i|
+      o = target.configuration_script_sources.find_or_build(i.id.to_s)
+      o[:description] = i.description
+      o[:name] = i.name
+
+      i.playbooks.each do |playbook_name|
+        # FIXME: its not really nice how I have to build a manager_ref / uuid here
+        p = target.playbooks.find_or_build("#{i.id}__#{playbook_name}")
```

Comment:
right, so it's better to have in unchanged in our db, so e.g. it you want to delete a playbook you know you need playbook.manager_ref and playbook.configuration_script_source.manager_ref

So you don't need to do a playbook.manager_ref.split("__")

I was proposing another key for pure unique check in https://github.com/ManageIQ/manageiq/issues/13806